lib/p2p/Pool.ts

```diff
@@ -846,7 +847,7 @@ class Pool extends EventEmitter {
   }
 
   /**
-   * Starts listening for incoming p2p connections on the configured host and port. If `this.listenPort` is 0 or undefined,
+   * Start listening for incoming p2p connections on specified hosts and ports. If `this.listenPort` is 0 or undefined,
```

Comment:
Proper style here is actually "Starts" instead of "Start" for javadoc so I've been trying to stick to that. Also it seems like we're not using `listenPort` any more, can drop it from this method description and maybe from the class altogether?

Reply:
`listenPort` is actually being used in `detectExternalIpAddress` I guess it's not being set correctly but somehow I feel like we need to set one of the ports from interfaces into `this.nodeState.addresses` ?

Reply:
`detectExternalIpAddress` indeed is not working. Did you see why? @rsercano

Reply:
do you mean for this branch @kilrau or totally? :)

Reply:
There's actually a mess in `Pool.ts` right now, since it's been a lot after @ImmanuelSegol 's commits there're lots of changes and I'm trying to figure them out to merge correctly now.